openedx_external_enrollments/tests/external_enrollments/tests_pathstream_external_enrollments.py

```diff
@@ -120,37 +139,73 @@ def test_download_file_init_s3_not_called(self):
             self.base._download_file,  # pylint: disable=protected-access
         )
 
+    def test_prepare_new_content(self):
+        """This test validates that _prepare_new_content accomplish
+        its task of sorting and encoding."""
+        content = [
+            'course_T1,1@mail,2021-06-29 16:53:00.456900,false\n',
+            'course_T2,2@mail,2021-06-29 16:51:00.456900,true\n',
+            'course_T1,1@mail,2021-06-29 16:50:00.456900,true\n',
+            'course_T2,2@mail,2021-06-29 16:52:00.456900,false\n',
+        ]
+        expected_result = (
+            b'course_T1,1@mail,2021-06-29 16:50:00.456900,true\n'
+            b'course_T2,2@mail,2021-06-29 16:51:00.456900,true\n'
+            b'course_T2,2@mail,2021-06-29 16:52:00.456900,false\n'
+            b'course_T1,1@mail,2021-06-29 16:53:00.456900,false\n'
+        )
+
+        result = self.base._prepare_new_content(content)  # pylint: disable=protected-access
+
+        self.assertEqual(result, expected_result)
+
     def test_upload_file(self):
         """This test validates that _upload file calls the method to update
         the S3 file with the right parameters."""
         s3_file = 'test.log'
         s3_bucket = 'test'
         self.base.client = Mock(spec=boto3.client('s3'))
+        self.base.client.put_object.return_value = {
+            'ResponseMetadata': {
+                'HTTPStatusCode': 200,
+            },
+        }
+        content = b'content'
+        content_md5 = 'mgNkuembtIDdJeHwKEyFVQ=='
         log = 'File [{}] was uploaded to S3 for [{}]'.format(s3_file, self.base.__str__())
 
         with LogCapture(level=logging.INFO) as log_capture:
-            self.base._upload_file()  # pylint: disable=protected-access
+            self.base._upload_file(content)  # pylint: disable=protected-access
 
             log_capture.check(
                 (self.module, 'INFO', log),
             )
 
-        self.base.client.upload_file.assert_called_with(
-            s3_file,
-            s3_bucket,
-            s3_file,
+        self.base.client.put_object.assert_called_with(
+            Bucket=s3_bucket,
+            Key=s3_file,
+            Body=content,
+            ContentMD5=content_md5,
         )
 
+    @ddt.data(None, b'', 'string', [], {})
+    def test_upload_file_with_wrong_file_content(self, content):
+        """When _upload_file is called without the content parameter or its value
+        is empty, it must raise an ExecutionError."""
+        self.base.client = Mock(spec=boto3.client('s3'))
+        with self.assertRaises(ExecutionError):
+            self.base._upload_file(content)  # pylint: disable=protected-access
+
     def test_upload_file_error(self):
```

Comment:
@Jacatove, is there a way to assert that the operation `put_object` has been executed?

Reply:
@ivanvgh I assert that the operation `put_object` is called with the right parameter in the test `test_upload_file`, but in case you meant asserting that `put_object` is not called. I will add that assertion.

Reply:
Same comment of line 87. @Jacatove

Reply:
Also, check the exception message. @Jacatove